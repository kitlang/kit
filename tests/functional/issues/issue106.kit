struct MyStruct {
    var a: Int;
}

function f(x: Option[MyStruct] = None) {
    return x;
}

function g(x: Option[MyStruct] = Option[MyStruct].None) {
    return x;
}

function main() {
    var s = f();
    if s.isNone() {
        puts("hi1");
    }

    var t = g();
    if t.isNone() {
        puts("hi2");
    }

    var u = f(Some(struct MyStruct));
    if u.isNone() {
        puts("hi3");
    }
}
