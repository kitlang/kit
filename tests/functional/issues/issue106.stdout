hi1
hi2
